### PR TITLE
Add radially compressed coordinates

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Coordinates suitable for visualizing large radii by compressing them logarithmically or inversely.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Domain/RadiallyCompressedCoordinates.hpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  struct Compression {
+    using type = CoordinateMaps::Distribution;
+    static constexpr Options::String help =
+        "Compression mode: 'Logarithmic' or 'Inverse'. If you use the same "
+        "mode that you used to distribute radial grid points then the "
+        "grid points will be distributed linearly in the radially compressed "
+        "coordinates.";
+  };
+  using options = tmpl::list<InnerRadius, OuterRadius, Compression>;
+  void pup(PUP::er& p) {
```
Comment (nilsdeppe) [inline src/Domain/RadiallyCompressedCoordinates.hpp]: Move to cpp?

File: src/Domain/RadiallyCompressedCoordinates.hpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    gsl::not_null<tnsr::I<DataType, Dim, CoordsFrame>*> result,
+    const tnsr::I<DataType, Dim, CoordsFrame>& coordinates, double inner_radius,
+    double outer_radius, CoordinateMaps::Distribution compression);
+template <typename DataType, size_t Dim, typename CoordsFrame>
+tnsr::I<DataType, Dim, CoordsFrame> radially_compressed_coordinates(
+    const tnsr::I<DataType, Dim, CoordsFrame>& coordinates, double inner_radius,
+    double outer_radius, CoordinateMaps::Distribution compression);
+/// @}
+
+struct RadiallyCompressedCoordinatesOptions {
```
Comment (nilsdeppe) [inline src/Domain/RadiallyCompressedCoordinates.hpp:73]: Please document

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
you can squash immediately

**Comment by nilsvu** [comment]
done 👍

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**